### PR TITLE
Consistent ConfigureAwait(false)

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

**Comment by matthchr** [comment]
Is there a plan to turn on the Analyzer rule which enforces this too to prevent regressions?

**Comment by danielmarbach** [comment]
Sorry was doing this on the plan and as soon as I had internet I pushed. Will make sure it properly compiles later today

**Review by heaths — COMMENTED** [review]
@pakrym almost all of this is track 1 code. What is the plan for package refresh in cases like this, and should analyzers be enabled for track 1 code - this analyzer, at least?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ internal sealed class CbsTokenProvider : ICbsTokenProvider
                                                   string appliesTo,
                                                   string[] requiredClaims)
         {
-            AccessToken token = await _credential.GetTokenAsync(new TokenRequest(requiredClaims), _cancellationToken);
+            AccessToken token = await _credential.GetTokenAsync(new TokenRequest(requiredClaims), _cancellationToken).ConfigureAwait(false);
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs]: Good catch.   This was initially overlooked, but was recently fixed as part of ongoing efforts and will ride along with an forthcoming Event Hubs PR.   That said, I've no objection to it coming in via this PR as well.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ internal sealed class CbsTokenProvider : ICbsTokenProvider
                                                   string appliesTo,
                                                   string[] requiredClaims)
         {
-            AccessToken token = await _credential.GetTokenAsync(new TokenRequest(requiredClaims), _cancellationToken);
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs]: This was originally written before we turned on the analyzers, but I can confirm that it doesn't appear to trigger any warnings or other notification so far as I can see.

[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs]: The analyzer doesn't work with ValueTasks there is an issue filed about it but I'm not sure if it's actively getting fixed.

**Review by isra-fel — APPROVED** [review]
Looks good for the management plane. Thank you 👍

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs (hunk 1 above)
Comment (danielmarbach) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs]: Rebased on top of latest changes

**Comment by danielmarbach** [comment]
Given that this touches quite a few points in the SDK is there anything I can do to speed up the review process to make sure I don't have to continuously rebase this PR?

Happy to make the reviewers' life easier if you need me to change anything. For example, I can also split up the one PR into multiple PRs per component if that is desired

**Comment by danielmarbach** [comment]
The build fails now with event hub CI operation canceled. I have no idea why this happened

**Comment by jsquire** [comment]
> The build fails now with event hub CI operation canceled. I have no idea why this happened

Triggering a re-run worked without issue.

Something seems to have caused the Linux tests to lose communication with the runner or there is a test with an intermittent case of not terminating.   I've not seen that before, locally or in CI, nor am I able to find a smoking gun in the logs.  I've taken a note to try and get some instrumentation in there to help diagnose if we run into it again.

**Comment by heaths** [comment]
These are good catches and we will take the PR. Were you running into any problems that you need 1 or more packages released?

**Comment by danielmarbach** [comment]
No I'm good. I did the investigation because of https://github.com/Azure/azure-sdk-for-net/issues/6976 and the question from alex.

**Pull request merged**